client/logmon/plugin.go

```diff
@@ -24,12 +24,13 @@ func LaunchLogMon(logger hclog.Logger) (LogMon, *plugin.Client, error) {
 	client := plugin.NewClient(&plugin.ClientConfig{
 		HandshakeConfig: base.Handshake,
 		Plugins: map[string]plugin.Plugin{
-			"logmon": NewPlugin(NewLogMon(hclog.L().Named("logmon"))),
+			"logmon": &Plugin{},
```

Comment:
Mind explaining why you want an empty Plugin struct here? I'd like to get more familiar with the plugin plumbing.

Reply:
Yeah of course! The plugin map is used to map the name to the plugin implementation. When go-plugin dispenses it uses the map to get the implementation. Then when you ask for a server and client, you get access to this:
https://github.com/hashicorp/nomad/blob/master/client/logmon/plugin.go#L59

Here we are asking go-plugin to give us a client to a remote plugin, so we actually do not need an implementation of the remote plugin! Instead we just want the struct that resolves the method to get a GRPC client that can talk to a remote version: https://github.com/hashicorp/nomad/blob/master/client/logmon/plugin.go#L67.

So in the old style, we were actually creating a NewLogMon locally which could have side affects where we didn't need to be!